app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicPlaybackService.java

```diff
@@ -462,4 +493,74 @@ public void onDestroy() {
     public IBinder onBind(Intent intent) {
         return mBinder;
     }
+
+    private class MediaSessionCallback extends MediaSession.Callback {
+
+        @Override
+        public boolean onMediaButtonEvent(Intent mediaButtonIntent) {
+            KeyEvent keyEvent = mediaButtonIntent.getParcelableExtra(Intent.EXTRA_KEY_EVENT);
+            Log.d(TAG, "onMediaButtonEvent in MediaSessionCallback called with event: " + keyEvent);
+            if (keyEvent != null) {
+                if (keyEvent.getAction() == KeyEvent.ACTION_UP) {
+                    Log.d(TAG, "onMediaButtonEvent: ACTION_UP pressed");
+                } else if (keyEvent.getAction() == KeyEvent.ACTION_DOWN) {
+                    Log.d(TAG, "onMediaButtonEvent: ACTION_DOWN pressed");
+                }
+                switch (keyEvent.getKeyCode()) {
+                    case KeyEvent.KEYCODE_MEDIA_PLAY:
+                        Log.d(TAG, "onMediaButtonEvent: KEYCODE_MEDIA_PLAY pressed");
+                        break;
+                    case KeyEvent.KEYCODE_MEDIA_PAUSE:
+                        Log.d(TAG, "onMediaButtonEvent: KEYCODE_MEDIA_PAUSE pressed");
+                        break;
+                    case KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE:
+                        Log.d(TAG, "onMediaButtonEvent: KEYCODE_MEDIA_PLAY_PAUSE pressed");
+                        break;
+                }
+            }
+            return super.onMediaButtonEvent(mediaButtonIntent);
+        }
+
+        @Override
+        public void onPlay() {
+            Log.d(TAG, "onPlay in MediaSessionCallback called");
+            play();
+        }
+
+        @Override
+        public void onPause() {
+            Log.d(TAG, "onPause in MediaSessionCallback called");
+            pause();
+        }
+
+        @Override
+        public void onSkipToNext() {
```

Comment:
Just ignoring these? Btw, you can use these callbacks as a way to communicate between the front end (playback fragment/activity/whatever) and this service rather than using a bound service to make calls.